src/login/LoginPage.jsx

```diff
@@ -76,7 +101,19 @@ const LoginPage = (props) => {
     if (loginErrorCode) {
       setErrorCode(prevState => ({ type: loginErrorCode, count: prevState.count + 1 }));
     }
-  }, [loginErrorCode]);
+    if (thirdPartyErrorMessage) {
+      setErrorCode(() => ({ type: TPA_AUTHENTICATION_FAILURE, context: {
+        errorMessage: thirdPartyErrorMessage,
+      }}));
+    }
+  }, [loginErrorCode, thirdPartyErrorMessage]);
```

Comment:
I know this will be working but can we please change the approach here? Since at initialization we defined this useState to have the initial value `{ type: '', count: ''}`, and here we are appending context to it as well. This does not make any sense code wise will lead to difficulty in debugging in the future? 

My suggestions would be to have some common state for `LoginFailure` component props and set errors in that if any. Or maybe we can stick to the old approach of conditionally rendering the `LoginFailure` component with third party error and loginError conditions?

Reply:
I am confused about it we need to discuss this